src/dom/get-visible-y-from-root.ts

```diff
@@ -0,0 +1,45 @@
+import { dom } from '../';
+import { domVectorf } from './dom-vectorf';
+
+function isFixed(element: HTMLElement) {
+  return dom.getStyle(element).position === 'fixed';
+}
+
+function getTransformedOffset(candidateElement: HTMLElement): number {
+  return candidateElement.offsetTop +
+      domVectorf.fromElementTransform(candidateElement).y;
+}
+
+function getVisibleDistanceFromRoot(
```

Comment:
Can you add usage examples?  Also consider using words ancestor as I think Dom classes use that as  a common term.